modules/client/src/controllers/TransferController.ts

```diff
@@ -180,12 +177,12 @@ export class TransferController extends AbstractController {
         boundResolve = this.resolveInstallTransfer.bind(null, res);
         this.listener.on(CFCoreTypes.EventName.INSTALL_VIRTUAL, boundResolve);
         this.listener.on(CFCoreTypes.EventName.REJECT_INSTALL_VIRTUAL, boundReject);
-        // this.timeout = setTimeout(() => {
-        //   this.cleanupInstallListeners(boundResolve, boundReject);
-        //   boundReject({ data: { appInstanceId: this.appId } });
-        // }, 5000);
+        this.timeout = setTimeout((): void => {
+          this.cleanupInstallListeners(boundResolve, boundReject);
+          boundReject({ data: { appInstanceId: this.appId } });
```

Comment:
wont reject properly:
```
// check app id
if (this.appId !== (msg as any).appInstanceId) {	    
      return;
}
```

Reply:
We shouldn't do it this way, using Promise.race seems a lot cleaner:

https://github.com/ConnextProject/indra/pull/466/files?file-filters%5B%5D=.ts#diff-d47b6d123be3db39095d99eea5357c64R178